<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html><head><title>R: Differential expression analysis of per-nucleotide coverage
tables.</title>
<meta http-equiv="Content-Type" content="text/html; charset=utf-8">
<link rel="stylesheet" type="text/css" href="R.css">
</head><body>
<h1> Differential expression analysis of per-nucleotide coverage
tables.
<img class="toplogo" src="../../../doc/html/logo.jpg" alt="[R logo]">
</h1>
<hr>
<div align="center">
<a href="../../../doc/html/packages.html"><img src="../../../doc/html/left.jpg" alt="[Up]" width="30" height="30" border="0"></a>
<a href="../../../doc/html/index.html"><img src="../../../doc/html/up.jpg" alt="[Top]" width="30" height="30" border="0"></a>
</div><h2>Documentation for package &lsquo;tornado&rsquo; version 1.0</h2>

<ul><li><a href="../DESCRIPTION">DESCRIPTION file</a>.</li>
</ul>

<h2>Help Pages</h2>


<table width="100%">
<tr><td width="25%"><a href="tornado-package.html">tornado-package</a></td>
<td>Tornado</td></tr>
<tr><td width="25%"><a href="find.mean.html">find.mean</a></td>
<td>optional helper function for getParams</td></tr>
<tr><td width="25%"><a href="find.mean.down.html">find.mean.down</a></td>
<td>helper function for getParams</td></tr>
<tr><td width="25%"><a href="find.mean.up.html">find.mean.up</a></td>
<td>helper function for getParams</td></tr>
<tr><td width="25%"><a href="find.sd.html">find.sd</a></td>
<td>helper function for getParams</td></tr>
<tr><td width="25%"><a href="get.numalts.html">get.numalts</a></td>
<td>helper function for getParams</td></tr>
<tr><td width="25%"><a href="getAnnotation.html">getAnnotation</a></td>
<td>download exon information for a given genome</td></tr>
<tr><td width="25%"><a href="getExons.html">getExons</a></td>
<td>find closest exon(s) to a genomic region</td></tr>
<tr><td width="25%"><a href="getLimmaInput.html">getLimmaInput</a></td>
<td>fit a linear model to each nucleotide</td></tr>
<tr><td width="25%"><a href="getParams.html">getParams</a></td>
<td>calculate parameters to use as input for HMM</td></tr>
<tr><td width="25%"><a href="getParams.failsafe.html">getParams.failsafe</a></td>
<td>helper function for getParams</td></tr>
<tr><td width="25%"><a href="getRegions.html">getRegions</a></td>
<td>generate list of regions, classify each as differentially expressed or not</td></tr>
<tr><td width="25%"><a href="getTstats.html">getTstats</a></td>
<td>calculate moderated t statistic for each nucleotide</td></tr>
<tr><td width="25%"><a href="gsubfn.html">gsubfn</a></td>
<td>helper function for read.csv.sql</td></tr>
<tr><td width="25%"><a href="last.html">last</a></td>
<td>get the last element</td></tr>
<tr><td width="25%"><a href="locfdrFit.html">locfdrFit</a></td>
<td>modified version of Efron's locfdr</td></tr>
<tr><td width="25%"><a href="makeDb.html">makeDb</a></td>
<td>create SQLite database from text file</td></tr>
<tr><td width="25%"><a href="match.funfn.html">match.funfn</a></td>
<td>Generic extended version of R match.fun</td></tr>
<tr><td width="25%"><a href="ostrapply.html">ostrapply</a></td>
<td>helper function for read.csv.sql</td></tr>
<tr><td width="25%"><a href="plotExon.html">plotExon</a></td>
<td>plot pipeline data/results for a given exon</td></tr>
<tr><td width="25%"><a href="plotGene.html">plotGene</a></td>
<td>plot pipeline data/results for a given gene</td></tr>
<tr><td width="25%"><a href="plotRegion.html">plotRegion</a></td>
<td>plot pipeline data/results for a given region</td></tr>
<tr><td width="25%"><a href="read.csv.sql.html">read.csv.sql</a></td>
<td>main workhorse of makeDb</td></tr>
<tr><td width="25%"><a href="sqldf.html">sqldf</a></td>
<td>helper function for read.csv.sql</td></tr>
<tr><td width="25%"><a href="strapply.html">strapply</a></td>
<td>helper function for read.csv.sql</td></tr>
<tr><td width="25%"><a href="supportedTables.html">supportedTables</a></td>
<td>print list of supported (downloadable) tables for a given genome</td></tr>
<tr><td width="25%"><a href="tornado-package.html">tornado</a></td>
<td>Tornado</td></tr>
</table>
</body></html>
